packages/amplify_datastore/example/lib/codegen/Post.dart

```diff
@@ -0,0 +1,215 @@
+/*
+*
+* Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:meta/meta.dart';
+import 'package:collection/collection.dart';
+
+import 'Blog.dart';
+import 'Comment.dart';
+
+/*
+type Post @model @key(name: "byBlog", fields: ["blogID"]) {
+  id: ID!
+  title: String!
+  rating: Int!
+  created: AWSDate!
+  blogID: ID!
+  blog: Blog @connection(fields: ["blogID"])
+  comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
+}
+ */
+
+@immutable
+class Post extends Model {
```

Comment:
Seems like there are a lot of redundant models in this PR. Can you just use the `testData` ones across the various modules?

Reply:
That's a bit tough and I don't know a way to do that without more research on this topic. 

The main issue is that the test models exist within the test folder which cannot be accessed from the separate example project.  In the opposite direction, the test is not able to access contents within the example folder.  This is likely because both are not packages so they aren't designed to be used elsewhere. 

I think I could create the codegen files within the package itself, but that would mean providing it within our public package which might be confusing to our end users.  

We could possibly create an entirely different package and include them as dependencies in our pubspec.yaml (as a dev dependency for tests and as a regular dependency for the example app)  

I see that as a possible solution.  However I don't see this as a major issue yet and I think we might want to put more time into finding a better solution before comitting more time to a specific solution.

Reply:
> We could possibly create an entirely different package and include them as dependencies in our pubspec.yaml (as a dev dependency for tests and as a regular dependency for the example app)

This is precisely what the Android code base does. We have a module called `testmodels`, that we don't publish, but that gets consumed by the unit and integration tests of several other modules.

I do recommend this approach.